test/rules/join_return_with_assignment.dart

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
```

Comment:
Could you rename this file to `join_return_with_assignment_test.dart` and then add it to `all.dart` too?

Reply:
Done.